File: leetcode/1.两数之和/1.java
class Solution {
    public int[] twoSum(int[] nums, int target) {
        Map<Integer, Integer> cnt = new HashMap<>();
        for(int i = 0; i < nums.length; i++){
            int ano = target - nums[i];
            if(cnt.containsKey(ano)){
                return new int[]{cnt.get(ano), i};
            }
            cnt.put(nums[i], i);
        }
        return new int[]{};
    }
}

File: leetcode/1.两数之和/1.cpp
class Solution {
public:
    vector<int> twoSum(vector<int>& nums, int target) {
        unordered_map<int, int> cnt;
        // vector<int> result;
        // if(nums.size() == 0 || nums.size() == 1)
        //     return {};
        // cnt.insert(nums[0]);
        for (int i = 0; i < nums.size(); i++) {
            int ano = target - nums[i];
            if (cnt.find(ano) != cnt.end()) {
                return { cnt[ano], i };
            }
            cnt[nums[i]] = i;
        }
        return {};
    }
};